### PR TITLE
marionette driver support added

### PR DESCRIPTION
**Pull request by pumano**

### CODE REVIEW

File: repository.json [hunk 1]
```diff
@@ -678,6 +678,34 @@
...
+            "version": "0.8.0",
+            "url": "https://github.com/mozilla/geckodriver/releases/download/v0.8.0/geckodriver-0.8.0-OSX.gz"
+        },
+        {
+            "name": "marionette",
+            "platform": "mac",
+            "bit": "64",
+            "version": "0.8.0",
+            "url": "https://github.com/mozilla/geckodriver/releases/download/v0.8.0/geckodriver-0.8.0-OSX.gz"
         }
```
Comment (lkwg82) [inline repository.json:702]: Formally this is correct. But see [maven plugin](https://github.com/webdriverextensions/webdriverextensions-maven-plugin/blob/master/src/main/java/com/github/webdriverextensions/Repository.java#L138). We have a fallback to 32bit for not found 64bit os. 

So for the ease of maintenance remove the almost identical entry for macos.

Comment (pumano) [inline repository.json:702]: I asked details about bits of MacOS release in marionette github page, they will tell details and then I change it, if it needed.

File: repository.json [hunk 2]
```diff
@@ -678,6 +678,34 @@
             "version": "1.9.7",
             "url": "https://bitbucket.org/ariya/phantomjs/downloads/phantomjs-1.9.7-windows.zip",
             "fileMatchInside": ".*/phantomjs.exe$"
+        },
+        {
+            "name": "marionette",
```
Comment (andidev) [inline repository.json]: Should't we call the driver geckodriver instead of marionette? According the geckodriver github page Marionette is the protocol name while the drivers name is geckodriver.

File: repository.json (hunk 1 above)
Comment (andidev) [inline repository.json:702]: @lkwg82 
Yes you are right and it needs to be handled similar to how the chromedriver is handled in the `detectBits(String driverName)` method in `Repository.java`. See https://github.com/webdriverextensions/webdriverextensions-maven-plugin/blob/master/src/main/java/com/github/webdriverextensions/Repository.java#L191-L193

e.g.

``` java
if (driverName.equals("marionette") && !isLinux()) {
    return "32";
}
```

Not sure how to handle the missing 32 bit linux driver though or maybe that is no issue, i think i need to think about it.

File: repository.json (hunk 2 above)
Comment (pumano) [inline repository.json]: OK I can fix it today and change it to geckodriver.

File: repository.json (hunk 1 above)
Comment (lkwg82) [inline repository.json:702]: Best would be to follow the strategy @pumano has suggested for fallbacks (chromedriver):

driver 32bit --> 32bit-driver-url
driver 64bit --> 32bit-driver-url

This would remove the code smells. But we need tests and this should be done in an different PR.

Comment (andidev) [inline repository.json:702]: The problem with that fallback solution is that it would not work when trying to detect bits for Windows 10 and the Internet Explorer driver. 

The alternatives for me seems to be to
- Remove the bit detection from the plugin and enforce the user to set it.
- Make the bit detection simple and always download the bit version of the OS 
- Somehow make it configurable through the repository.json

But that's another PR as you say.

I'll merge this PR and add the if case myself and then do a release ASAP.

File: repository.json (hunk 2 above)
Comment (pumano) [inline repository.json]: I need change marionette to geckodriver in this PR or in next?

Comment (andidev) [inline repository.json]: Please change that in this if you have time. Otherwise I'll do it.

Comment (pumano) [inline repository.json]: Ok I will change it now.

File: repository.json (hunk 1 above)
Comment (pumano) [inline repository.json:702]: Man from geckodriver community told me: "Releases will be 64bit only. Going forward there will be no Firefox releases for OSX < 10.9 Since 10.9 won't run on 32 bit CPUs there is no point in making a 32bit geckodriver release." It means I need to delete 32 bit mac release of geckodriver. I can do it soon (maybe today or tomorrow).

Comment (lkwg82) [inline repository.json:702]: @andidev 

> Remove the bit detection from the plugin and enforce the user to set it.

This does not work. I observe many dev workstations to be linux/mac and ci systems like jenkins to be linux. It is not sure to have the same bit combination on both systems.

I'd prefer a special case handling for the special case "iedriver".

What about that?

Comment (pumano) [inline repository.json:702]: Some workstations can be windows too, because they also popular. I have windows 10 workstation for selenium and mac for appium

Comment (andidev) [inline repository.json:702]: Yes you are right and it looks like it works without the bit detection logic for the other drivers. @lkwg82 Sorry this is probably what your last PR was about but it looks like I got a bit confused.

Comment (lkwg82) [inline repository.json:702]: No sorry! ;) We can move on!

**Comment by andidev** [comment]
Closing this since it was added in the new repo for version 3.0 webdriverextensions-maven-plugin-repository at
https://github.com/webdriverextensions/webdriverextensions-maven-plugin-repository/blob/master/repository-3.0.json

Thanx!
